### ISSUE TITLE
SSR not working as expected in with-apollo example

### ISSUE THREAD

**Issue by belgianwolfie** [zeit/next.js#8745]
# Examples bug report

## Example name

with-apollo (https://github.com/zeit/next.js/tree/canary/examples/with-apollo)

## Describe the bug

As soon as I replace the graphql uri inside 'lib/apollo.js' to be my own (my uri is a graphql endpoint provided by Drupal8) when I reload the 'homepage' of the app I get the "loading" indicator every single time. (altho sometimes I see my (static) data flash really fast before 'loading' comes on)

Another difference between my own 'version' vs the example is that I am not using 'skip' or 'first' allPostQueryVars as I only have 130 items max and I need them all at once (for searching).

Notable differences in Network (tab) activity:
clean with-apollo example from github:
- '0.js' gets loaded
- right after 'webpack-hmr?page=/' gets loaded and thats it

when I change to my URI:
- '0.js' gets loaded
**- right after another call to graphql (my drupal) happens which causes the reload**
- 'webpack-hmr?page=/' gets loaded

In my console I get a **Warning** '**Warning: Text content did not match.**'
`Warning: Text content did not match. Server <this shows my entire data for the PostList component> Client 'Loading'`

## To Reproduce

1. clone the with-apollo repo at https://github.com/zeit/next.js/tree/canary/examples/with-apollo
2. replace the uri in lib/apollo.js with your own (I'm using Drupal8 as my graphql server)
3. reload / and see 'Loading' always show up

## Expected behavior

While the data is getting loaded eventually, it shouldn't since ssr is default to true and the example clearly states on the 'homepage' infobox: " This example shows how to fetch all initial apollo queries on the server. If you <a href='/'>reload</a> this page you won't see a loader since Apollo fetched all needed data on the server."

## Screenshots

no screenshots

## System information

- Chrome browser
- happens on localhost and NOW deployed 
- Next latest version

## Additional context

Since I really 'only' (I obviously also needed to change the fieldnames) changed the uri I'm guessing this has something to do with the cache-settings for my graphql endpoint?
But this is just a wild guess

**Comment by belgianwolfie** [comment]
**update:**
I already spent an entire friday evening and saturday on the issue before posting here.
Obviously - right after posting here I realised something else.

My graphql looks like this and takes a variable parameter:
`query allPosts {
  nodeQuery(limit:99999,filter: {conditions: [
    {operator: EQUAL, field: "type", value: ["Event"]},
    {operator: GREATER_THAN_OR_EQUAL, field: "field_datetime", value: ["${**customoffset**}"]}
    ]},
    sort: [{`

If I leave that 'GREATER_THAN_OR_EQUAL' with the customoffset param out, it does not do the hard reload anymore. Makes sense obviously. Graphql is now cacheable I guess.
Still get a Warning about text content did not match, but not against 'Loading' anymore

**Comment by giorgenes** [comment]
@belgianwolfie not sure it's related to your problem. I've been trying the apollo graphql sample as well and one thing I've noticed is the requests work fine when done with SSR, but when it tried to fetch directly from the browser it fails because it doesn't respect the configured URL. It tries to fetch from `host/graphql`. I believe this is as design but I had to put an nginx proxy in front to make it work.

I'm also having other issues with the page not loading some times in some scenarios.
